Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs

```diff
@@ -0,0 +1,303 @@
+        {
+            _ytdlFound = false;
+            #if UNITY_EDITOR_WIN
+            string[] splitPath = Application.persistentDataPath.Split('/', '\\');
+            
+            // Check for yt-dlp in VRC application data first
+            _ytdlpPath = string.Join("\\", splitPath.Take(splitPath.Length - 2)) + "\\VRChat\\VRChat\\Tools\\yt-dlp.exe";
+            #elif UNITY_EDITOR_LINUX
+            _ytdlpPath = "/usr/bin/yt-dlp";
+            #endif
+            if (!File.Exists(_ytdlpPath)) 
+            {
+                // Check the local path (in the Assets folder)
+                _ytdlpPath = _localYtdlpPath;
+            }
+
+            if (!File.Exists(_ytdlpPath))
+            {
+                #if UNITY_EDITOR_WIN
+                // Offer to download yt-dlp to the AudioLink folder
+                bool doDownload = EditorUtility.DisplayDialog("[AudioLink] Download yt-dlp?", "AudioLink could not locate yt-dlp in your VRChat folder.\nDownload to AudioLink folder instead?", "Download", "Cancel");
+                if(doDownload)
+                    DownloadYtdlp();
+
+                if(!Application.isPlaying)
+                    EditorApplication.ExitPlaymode();
+                
+                #elif UNITY_EDITOR_LINUX
+                    EditorUtility.DisplayDialog("[AudioLink] Missing yt-dlp", "try \"sudo pacman -S yt-dlp\"", "ok");
+                #endif
+            }
+
+            if (!File.Exists(_ytdlpPath)) 
+            {
+                // Still don't have it, no dice
+                Debug.LogWarning("[AudioLink] Unable to find yt-dlp");
+                return;
+            }
+            else
+            {
+                // Found it
+                _ytdlFound = true;
+                Debug.Log($"[AudioLink] Found yt-dlp at path '{_ytdlpPath}'");
+            }
+        }
+
+        /// <summary> Resolves a URL to one usable in a VideoPlayer. </summary>
+        /// <param name="url">URL to resolve for playback</param>
+        /// <param name="resolution">Resolution (vertical) to request from yt-dlp</param>
+        public static string Resolve(string url, int resolution)
+        {
+            if(!_ytdlFound)
+            {
+                Debug.LogWarning($"[AudioLink] Unable to resolve URL '{url}' : yt-dlp not found");
+                return null;
+            }
+
+            Process ytdlProcess = new Process();
+
+            ytdlProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
+            ytdlProcess.StartInfo.CreateNoWindow = true;
+            ytdlProcess.StartInfo.UseShellExecute = false;
+            ytdlProcess.StartInfo.RedirectStandardOutput = true;
+            ytdlProcess.StartInfo.FileName = _ytdlpPath;
+            ytdlProcess.StartInfo.Arguments = $"--no-check-certificate --no-cache-dir --rm-cache-dir -f \"mp4[height<=?{resolution}]/best[height<=?{resolution}]\" --get-url \"{url}\"";
+
+            Debug.Log($"[AudioLink] Attempting to resolve URL '{url}'");
```

Comment:
Is there really a reason to spam the log in the good path..

Reply:
Silence is golden

Reply:
Removed, resolved in 631f888e53bf119a63570ed455611d1e38ef151d